app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -3061,7 +3082,19 @@ class BrowserTabViewModel @Inject constructor(
         }
 
         viewModelScope.launch(dispatchers.main()) {
-            command.value = OnPermissionsQueryResponse(response)
+            command.value = SendResponseToJs(response)
+        }
+    }
+
+    private fun screenLock(featureName: String, method: String, id: String, data: JSONObject) {
+        viewModelScope.launch(dispatchers.main()) {
+            command.value = ScreenLock(JsCallbackData(data, featureName, method, id))
+        }
+    }
+
+    private fun screenUnlock() {
+        viewModelScope.launch(dispatchers.main()) {
+            command.value = ScreenUnlock
```

Comment:
Can we feature flag both screenLock and screenUnlock using a kill-switcg in `AndroidBrowserConfig`?
Something like.

```kotlin
    private fun screenUnlock() {
        viewModelScope.launch(dispatchers.io()) {
            if (andoidBrowserConfig.screenLock().isEnabled()) {
                withContext(dispatchers.main()) {
                    command.value = ScreenUnlock
                }
            }
        }
    }
```

I think this change is high potential to cause a regression and I want to avoid having to hotfix anything like we had recently with similar/related changes.

Reply:
We will have a feature flag on the C-S-S side: https://github.com/duckduckgo/content-scope-scripts/blob/nastia/screen_lock/src/features/web-compat.js#L114 so it won't send messages for either screenLock nor screenUnlock. Do you think we need an additional feature flag on the native side?

Reply:
I'd like to have a kill-switch in native. We've had issues with feature flags in JS before.